index.js

```
@@ -0,0 +1,161 @@
+
+var ConfigLoader = module.exports = Class('ConfigLoader', {
+
+    constructor: function ConfigLoader(storeName, directory, options) {
+
+        var defaults = {
+            environment: process.env.NODE_ENV || 'development',
+            instance: process.env.NODE_INSTANCE || '',
+            configDirectory: process.env.NODE_CONFIG_DIR || 'config',
+            defaultConfig: process.env.NODE_CONFIG_DEFAULT || 'default',
+            directory: directory,
+            envVarPrefixes: [
+                'NODE_CONFIG_',
+                'npm_package_config_',
+            ]
+        };
+        this.options = objectAssign(defaults, options || {});
+        this.store = ConfigLoader.stores[storeName] = {
+            environment: this.options.environment,
+            instance: this.options.instance,
+            directory: this.options.directory,
+        };
+    },
+
+    load: function load(callback) {
+        return this.readConfigDirectory(this.pathName)
+            .then(function (fileNames) {
+                return fileNames.filter(this.isValidFile, this);
+            }.bind(this))
+            .then(function (validFiles) {
+                return Promise.all([
+                    this.store,
+                    this.readConfigFile(validFiles, this.options.defaultConfig),
+                    this.readConfigFile(validFiles, this.options.defaultConfig, this.options.instance),
+                    this.readConfigFile(validFiles, this.options.environment),
+                    this.readConfigFile(validFiles, this.options.environment, this.options.instance)
+                ]);
+            }.bind(this)).then(function (configs) {
+                if (typeof callback === 'function') {
+                    return callback(this.merge(configs));
+                }
+                return this.merge(configs);
+            }.bind(this))
+            .catch(function (error) {
+                console.warn(error.stack || error);
+            });
```

This needs to be taken out. It swallows errors and does not let developers properly inspect error states. We should expect devs to `catch` in their impl.